src/NServiceBus.Core/Transports/ErrorContext.cs

```diff
@@ -22,7 +22,12 @@ public class ErrorContext
         /// <summary>
         /// Number of immediate processing attempts.
         /// </summary>
-        public int NumberOfDeliveryAttempts { get; }
+        public int NumberOfImmediateDeliveryAttempts { get; }
+
+        /// <summary>
+        /// Number of delayed processing attempts.
+        /// </summary>
+        public int NumberOfDelayedDeliveryAttempts { get; }
```

Comment:
Currently the semantics of this field is `NumberOfTheNextDeleyedDeliveryIfWePerformIt`. So from the perspective of retry policy if no retries have been performed this value will be 1. I think this should rather be `NumberOfFailedDelayedDeliveryAttempts` to be consistent with immediate retries.